xdsl/dialects/stencil.py

```diff
@@ -572,6 +581,10 @@ class BufferOp(IRDLOperation):
     temp: Operand = operand_def(TempType)
     res: OpResult = result_def(TempType)
 
+    def __init__(self: IRDLOperation, temp: SSAValue | Operation):
+        temp = SSAValue.get(temp)
```

Comment:
I don't think you need to do this here?

Reply:
I do, to some definition of "need", to access `.typ`